### PR TITLE
Fix false-positive for unnecessary-ellipsis checker on comparisons

### LINKED ISSUE #6071: False positive `unnecessary-ellipsis` when multiple conditions in `if` statement
### Bug description

```python
# test.py
def func(val1, val2):
   if val1 is not ... and val2:
        pass
```

Above code raises error, but the following does not:
```python
def func(val1, _):
   if val1 is not ...:
        pass
```

So, it probably is something to do with multiple conditions here.

### Configuration

_No response_

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:2:18: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)
```

### Expected behavior

No false-positives

### Pylint version

```shell
pylint 2.13.4
astroid 2.11.2
Python 3.10.0 (default, Oct 17 2021, 20:19:40) [Clang 12.0.5 (clang-1205.0.22.9)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by skshetry**
Closes #6071.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Fix false positive for `unnecessary-ellipsis` checker in comparisons.

Closes #6071.

### CODE REVIEW

**Review by DanielNoord — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
Maybe we should switch from disable list to enable list for this checker, we already had 3 issues with false positives for it. I.e. we list nodes where it should be raised instead of listing nodes where it should not be raised.

**Comment by mbyrnepr2** [comment]
> Maybe we should switch from disable list to enable list for this checker, we already had 3 issues with false positives for it. I.e. we list nodes where it should be raised instead of listing nodes where it should not be raised.

I'm OK with this idea. I think it's the pragmatic thing to do. In hindsight the scope of this checker is quite complex. Perhaps we can limit the scope to functions and classes which define a docsting followed by an ellipsis, and then expand the scope from there as desired.

**Comment by mbyrnepr2** [comment]
> > Maybe we should switch from disable list to enable list for this checker, we already had 3 issues with false positives for it. I.e. we list nodes where it should be raised instead of listing nodes where it should not be raised.
> 
> I'm OK with this idea. I think it's the pragmatic thing to do. In hindsight the scope of this checker is quite complex. Perhaps we can limit the scope to functions and classes which define a docstring followed by an ellipsis, and then expand the scope from there as desired.

Here is a MR with the more narrow scope for the checker: https://github.com/PyCQA/pylint/pull/6081; If this is ok with everyone perhaps we can review/merge that + the contents of the tests from the current MR? Otherwise please feel free to make any changes required in the current MR and ignore the one I've created.

File: ChangeLog [hunk 1]
```diff
@@ -48,7 +48,6 @@ Release date: TBA
   * A body which contains an ellipsis ``nodes.Expr`` node & at least one other statement.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:50]:
```suggestion


```

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you for the contribution @skshetry , the fix itself disappeared because we took a safer /more conservative approach, but the tests will stay forever :)

**Pull request merged**